bears/general/IndentationBear.py

```diff
@@ -153,6 +182,64 @@ def run(self,
                     affected_code=(diff.range(filename),),
                     diffs={filename: diff})
 
+    def _get_indent_keywords_range(self,
+                                   file,
+                                   filename,
+                                   indent_keywords,
+                                   ranges):
+        keyword_pos = {}
+        for keyword in indent_keywords:
+            keyword_pos[keyword] = (tuple(get_valid_sequences(file,
+                                                              keyword,
+                                                              ranges,
+                                                              keyword=True)))
+        keyword_range = []
+        for keyword in keyword_pos:
+            for position in keyword_pos[keyword]:
+                start = position
+                end = AbsolutePosition(file, position.position + len(keyword))
+                for encapsulator in ranges["encapsulators"]:
```

Comment:
This is kinda ugly but there is not much I can think of right now. It is possible to make a list comprehensions of end positions and take the last one, but you don't gain much by that.

Maybe try to determine the values for `end` and `indent` in their own functions to keep this block simpler?

at least add some new lines and comments to make this easier to read.